bin/preview.py

```diff
@@ -1,10 +1,9 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-from flask import Flask, redirect, send_from_directory, url_for
+from flask import Flask, redirect, send_from_directory, url_for # pylint: disable=E0401
```

Comment:
were you getting an import error ?
`requirements-dev.txt` contains flask.

Although I have no issue with adding this disable

Reply:
Indeed. Upon removing it, `pre-commit run -a` returns:
```
[sic]
pylint...................................................................Failed
- hook id: pylint
- exit code: 2

************* Module preview
bin\preview.py:4:0: E0401: Unable to import 'flask' (import-error)

-------------------------------------------------------------------
Your code has been rated at 9.92/10 (previous run: 10.00/10, -0.08)
[sic]
```

Note that `pylint bin` outputs:
```

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

So this is a `pre-commit` quirk. My guess is that they only install `requirements.txt`, instead of also `requirements-dev.txt`.

Reply:
thank you for adding the output 🙏 great documentation.

If you're curious to try it out, there is an option in pre-commit to customise requirements with https://pre-commit.com/#config-additional_dependencies and you can pass python libraries there (e.g. flask).

Reply:
I've tried it out briefly. In particular, I modified the `pylint` part fo the pre-commit-config.yaml to be this:
```
-   repo: https://github.com/PyCQA/pylint
    rev: pylint-2.7.2
    hooks:
    -   id: pylint
        additional_dependencies: [flask]
        exclude: pyfixdoc
```
And then removed the `# pylint: disable` after the flask import.

Now, the `pre-commit` also passes.

However, there are more dependencies that have `# pylint: disable=E0401` to avoid pre-commit complains. For example, after selenium imports from `bin/vis_doc.py`: https://github.com/DerwenAI/pytextrank/blob/15166621a7cc9fab58181349c594269b087d4f6f/bin/vis_doc.py#L4-L5

So, rather than half-assing this, we can go two routes:
* Place `pylint: disable=E0401` after all imports of dev-only dependencies. This is how it's currently done.
* Remove all `pylint: disable=E0401`, and tell `pre-commit` to use those additional dependencies.

Reply:
Yes let's stick with option 1 for now 👍 .

Reply:
Wonderful. I'll resolve this conversation, then. 👍